### ISSUE TITLE
Override for gbif taxonomy in make_taxonomy

### ISSUE THREAD

**Issue by Calamanthus** [Acanthiza/envClean#18]
We need some sort of override to deal with cases where the gbif taxonomic backbone is allocating the wrong taxonomy. 

This includes the the Hooded Plover (old name) being allocated to Red-necked Phalarope:
rgbif::name_backbone_checklist(c("Thinornis rubricollis","Phalaropus lobatus"))
A tibble: 2 × 25
  usageKey acceptedUsageKey scientificName        canonicalName rank  status confidence matchType kingdom phylum order family genus species kingdomKey phylumKey
<int>            <int> <chr>                 <chr>         <chr> <chr>       <int> <chr>     <chr>   <chr>  <chr> <chr>  <chr> <chr>        <int>     <int>
1  4849941          5739290 Thinornis rubricolli… Thinornis ru… SPEC… SYNON…         98 EXACT     Animal… Chord… Char… Scolo… Phal… Phalar…          1        44
2  5739290               NA Phalaropus lobatus (… Phalaropus l… SPEC… ACCEP…         99 EXACT     Animal… Chord… Char… Scolo… Phal… Phalar…          1        44

Also, another example I found recently where Cassia nemophila var. coriacea was being allocated to a Brazilian species in a different genus (Chamaecrista coriacea), instead of to Senna artemisioides:

rgbif::name_backbone_checklist(c("Chamaecrista coriacea","Cassia nemophila","Cassia nemophila var. coriacea"))
A tibble: 3 × 25
  usageKey scientificName       canonicalName rank  status confidence matchType kingdom phylum order family genus species kingdomKey phylumKey classKey orderKey
     <int> <chr>                <chr>         <chr> <chr>       <int> <chr>     <chr>   <chr>  <chr> <chr>  <chr> <chr>        <int>     <int>    <int>    <int>
1  2950087 Chamaecrista coriac… Chamaecrista… SPEC… ACCEP…         99 EXACT     Plantae Trach… Faba… Fabac… Cham… Chamae…          6   7707728      220     1370
2  5357114 Cassia nemophila A.… Cassia nemop… SPEC… SYNON…         97 EXACT     Plantae Trach… Faba… Fabac… Senna Senna …          6   7707728      220     1370
3  5357116 Cassia nemophila va… Cassia nemop… VARI… SYNON…         98 EXACT     Plantae Trach… Faba… Fabac… Cham… Chamae…          6   7707728      220     1370

The above example came out as having 100% of it's national AOO and EOO in the current study area, because of the erroneous attribution to a Brazilian species, when it should be an ultra common plant in Senna artemisioides. Interestingly, as seen above, without the variety (i.e. just 'Cassia nemophila') gbif attributes the taxonomy correctly.

**Comment by Calamanthus** [comment]
Another slightly different example...
Gypsophila australis was identified by red as having 100% of its range in the current study area, and it is accepted by the gbif taxonomic backbone, but is regarded as a synonym to Gypsophila tubulosa by Flora of Australia and ALA, which is identified in bdbsa as a weed. GBIF does not recognise G.tubulosa and assigns it to the family level, so it cannot be a straight taxonomy fix. It may be easiest to just attribute this to genus (which is accepted) and effectively remove it from the data, as I can think of lots of problems with other potential solutions. In this case it won't matter if the taxa is lost as it is a weed, but I just hope there isn't a similar example for a species that is indigenous. I hate this stuff! I'll start compiling these examples into a fixes table.

**Comment by Acanthiza** [comment]
I've added in a `taxonomy_overrides` argument to `make_taxonomy`. It follows a similar form to `taxonomy_fixes` but is implemented differently (via left_join) due to the string-to-find and the string-to-replace being in different columns. However, another tack-on-fix at this point highlights the precarious nature of our current taxonomy workflow. I've changed the appropriate code in envPIA, but envClean::taxonomy_overrides currently only deals with the Hooded Plover issues. I'll leave this open until it has been tested more thoroughly.

**Comment by Calamanthus** [comment]
Great, thanks. I'll add the other cases above. Where is envClean::taxonomy_overrides? I can't see it under ~/packages/envClean...

**Comment by Acanthiza** [comment]
It's in the taxonomy_fixes.R file (in data-raw).

**Comment by Calamanthus** [comment]
Ok, I was thinking it was its own script

**Comment by Calamanthus** [comment]
Just looking at the code...if this is just changing the taxa in the lutaxa result, then the best key will presumedly not be relevant to the updated taxa. Do we want an override field in lutaxa to flag that the gbif taxonomy has been changed and the best key is no longer relevant?

**Comment by Acanthiza** [comment]
The link to the taxonomic hierarchy to use (in taxa$taxonomy) is via the `taxa` column. The key wasn't being used. I've now removed the key from `lutaxa` output. I've tried to think of a way to make these fixes earlier so that the correct other attributes 
(e.g. the status, matchtype and rank) come through, but haven't been able to work out how to implement it. So yes, perhaps worth flagging that those fields may be incorrect in the lutaxa results if a 'fix' has been made.
